server/bleep/src/webserver/intelligence.rs

```diff
@@ -335,14 +315,60 @@ pub(super) async fn token_value(
     Ok(json(TokenValueResponse { range, content }))
 }
 
+pub async fn get_token_info(
+    params: TokenInfoRequest,
+    repo_ref: &RepoRef,
+    indexes: Arc<Indexes>,
+    source_doc: &ContentDocument,
+    all_docs: &Vec<ContentDocument>,
+    context_before: Option<usize>,
+    context_after: Option<usize>,
+) -> anyhow::Result<Vec<FileSymbols>> {
+    let source_document_idx = all_docs
+        .iter()
+        .position(|doc| doc.relative_path == source_doc.relative_path)
+        .ok_or(anyhow::anyhow!("invalid language"))?;
+
+    let snipper =
+        Some(Snipper::default().context(context_before.unwrap_or(0), context_after.unwrap_or(0)));
+
+    let ctx: CodeNavigationContext<'_, '_> = CodeNavigationContext {
+        token: Token {
+            relative_path: params.relative_path.as_str(),
+            start_byte: params.start,
+            end_byte: params.end,
+        },
+        all_docs,
+        source_document_idx,
+        snipper,
+    };
+
+    let data = ctx.token_info();
+    if data.is_empty() {
```

Comment:
@nerdypepper 

`data` here is empty, so we are using search_nav (tantivy search). When `token_info` returns a non empty vector? should it be faster than `search_nav`?